exporter/factory_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"testing"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/models"
+)
+
+type ExampleExporterFactory struct {
+}
+
+// Type gets the type of the Exporter config created by this factory.
+func (f *ExampleExporterFactory) Type() string {
+	return "exampleexporter"
+}
+
+// CreateDefaultConfig creates the default configuration for the Exporter.
+func (f *ExampleExporterFactory) CreateDefaultConfig() models.Exporter {
+	return nil
+}
+
+// CreateTraceExporter creates a trace exporter based on this config.
+func (f *ExampleExporterFactory) CreateTraceExporter(logger *zap.Logger, cfg models.Exporter) (consumer.TraceConsumer, StopFunc, error) {
+	return nil, nil, nil
+}
+
+// CreateMetricsExporter creates a metrics exporter based on this config.
+func (f *ExampleExporterFactory) CreateMetricsExporter(logger *zap.Logger, cfg models.Exporter) (consumer.MetricsConsumer, StopFunc, error) {
+	return nil, nil, nil
+}
+
+func TestRegisterExporterFactory(t *testing.T) {
+	f := ExampleExporterFactory{}
+	err := RegisterExporterFactory(&f)
+	if err != nil {
```

Comment:
q: why not `require.NotNil` (and related below)? I think we should make one of the two practices the standard on the repo. I personally don't have much of a preference, but I like your argument about `require` and `assert` making the tests shorter.

Reply:
These are not new tests, I just moved them. They were written before testify was added to the codebase. I agree we should try to follow a standard practice, but I don't want to touch the tests in this PR that is about something else.

Reply:
👍